src/validator/task.rs

```diff
@@ -790,11 +793,19 @@ fn connect_block(
             }
             None => (),
         };
-        let () = handle_m8(
+        if let Some(()) = handle_m8(
             transaction,
             &accepted_bmm_requests,
             &prev_mainchain_block_hash,
-        )?;
+        )
+        .inspect_err(|err| {
+            tracing::error!("Unable to handle M8 BMM request: {:?}", err);
```

Comment:
This is unnecessary. `ConnectBlockError` already has an explicit error message for M8s, and should be logged later by a handler in `main.rs`

```
#[derive(Debug, Error)]
enum ConnectBlockError {
    ...
    #[error("Error handling M8")]
    M8(#[from] HandleM8Error),
   ...
}
```